iib/workers/tasks/build.py

```diff
@@ -1151,7 +1077,25 @@ def handle_add_request(
             'index.Dockerfile',
         )
 
+        present_bundles_pull_spec = [bundle['bundlePath'] for bundle in present_bundles]
+        deprecation_bundles = get_bundles_from_deprecation_list(
+            present_bundles_pull_spec + resolved_bundles, deprecation_list
+        )
+
         arches = prebuild_info['arches']
+        if deprecation_bundles:
+            arch = 'amd64' if 'amd64' in arches else arches[0]
```

Comment:
`BundleDeprecation` uses bundle_id so keeping it consistent. If you feel strongly about it, I can change it

Reply:
This is how the merge-index-image does it when deprecating bundles. So I used this approach. I can change it if you want. Doesn't make much of a difference :)

Reply:
I think it's important to be consistent because we're matching bundles by digest. If our API endpoints are not using the same logic, there's a risk that something that should be deprecated doesn't. Can we have a follow up PR/ticket for this? I don't think we need to address it as part of this PR.

Reply:
ack, filed https://projects.engineering.redhat.com/browse/CLOUDDST-4901